app/subscriber/src/components/sidebar/advanced-search/utils/queryToState.ts

```diff
@@ -0,0 +1,27 @@
+import { fromQueryString } from 'tno-core';
+
+/** Function that returns advanced search filter state allowing the the inputs to stay in sync when the user is navigating back */
+export const queryToState = (queryString: string, setAdvancedSearch: (x: any) => void) => {
```

Comment:
I would lean to a more simple function that takes the query string and returns an object.  Then you can keep the state related function out of the parameters.  Keeping functions single purpose is better and easier to maintain.